Data Prep Code.Rmd

```diff
@@ -0,0 +1,317 @@
+
+# Conduct Kruskal-Wallis test
+kruskal_test <- kruskal.test(kept ~ treatment_group, data = df %>% filter(game_type == "Standard Dictator Game"))
+
+# Store the p-value for use in the plot annotation
+p_value <- round(kruskal_test$p.value, 4)
+
+# Create an annotation text with the Kruskal-Wallis result
+annotation_text <- paste("Kruskal-Wallis, p =", p_value, ". Note. Confidence intervals on means are at 95%.")
+print(annotation_text)
+
+# Plot the bar graph with sorted bars
+ggplot(standard_dictator_game_data, aes(x = treatment_group, y = Mean_Got, fill = treatment_group)) +
+  geom_bar(stat = "identity", width = 0.5, alpha = 0.5) +
+  geom_errorbar(aes(ymin = Mean_Got - SE, ymax = Mean_Got + SE), width = 0.2, color = "black") +
+  scale_fill_manual(values = c("C" = "red", "TD" = "green", "MD" = "blue")) +
+  labs(title = "Figure 2: Standard Dictator Game", y = "Mean of Got", x = NULL) +
+  theme_minimal() +
+  theme(plot.title = element_text(hjust = 0.5), legend.position = "none")
+
+```
+
+
+```{r Figure 3, echo=FALSE}
+# Filter the dataset for the Bully Dictator Game
+bully_dictator_game_data <- df %>%
+  filter(game_type == "Bully Dictator Game") %>%
+  mutate(treatment_group = case_when(
+    moc == 0 ~ "C",
+    moc == 1 ~ "TD",
+    moc == 2 ~ "MD"
+  )) %>%
+  group_by(treatment_group) %>%
+  summarise(
+    Mean_Got = round(mean(kept, na.rm = TRUE),2),
+    SD = round(sd(kept, na.rm = TRUE),2),
+    SE = round(sd(kept, na.rm = TRUE)/sqrt(n()),2) # Standard error
+  ) %>%
+  ungroup() %>%
+  # Arrange the data in descending order and set the factor levels based on this order
+  arrange(desc(Mean_Got)) %>%
+  mutate(treatment_group = factor(treatment_group, levels = unique(treatment_group)))
+
+# Print the adjusted data
+print(bully_dictator_game_data)
+
+
+# Explain what the Kruskal-Wallis test is applied on
+cat("Performing Kruskal-Wallis test to compare the distributions of 'kept' across different treatment groups within the Bully Dictator Game.\n")
+
+# Conduct Kruskal-Wallis test
+kruskal_test <- kruskal.test(kept ~ treatment_group, data = df %>% filter(game_type == "Bully Dictator Game"))
+
+# Store the p-value for use in the plot annotation
+p_value <- round(kruskal_test$p.value, 4)
+
+# Create an annotation text with the Kruskal-Wallis result
+annotation_text <- paste("Kruskal-Wallis, p =", p_value, ". Note. Confidence intervals on means are at 95%.")
+print(annotation_text)
+
+# Plot the bar graph with sorted bars
+ggplot(bully_dictator_game_data, aes(x = treatment_group, y = Mean_Got, fill = treatment_group)) +
+  geom_bar(stat = "identity", width = 0.5, alpha = 0.5) +
+  geom_errorbar(aes(ymin = Mean_Got - SE, ymax = Mean_Got + SE), width = 0.2, color = "black") +
+  scale_fill_manual(values = c("C" = "red", "TD" = "green", "MD" = "blue")) +
+  labs(title = "Figure 3: Bully Dictator Game", y = "Mean of Got", x = NULL) +
+  theme_minimal() +
+  theme(plot.title = element_text(hjust = 0.5), legend.position = "none")
+  

```

Comment:
Please use separate chunks for printing text/data/plots and write a comments as a plain text between those chunks.

Reply:
Done